api/document/tree.py

```diff
@@ -125,45 +219,50 @@ def nested_set_renumber(self, left=1, bulk_create=True):
         if bulk_create:
             self._bulk_create()
 
-    def _bulk_create(self):
+    def _bulk_create(self) -> None:
         DocNode.objects.bulk_create(node.model for node in self.walk())
 
-    def next_sort_order(self):
+    def next_sort_order(self) -> int:
         return self.tree.out_degree(self.identifier)
 
-    def parent(self):
+    def parent(self: T) -> Optional[T]:
         predecessors = list(self.tree.predecessors(self.identifier))
         if predecessors:
             return type(self)(self.tree, predecessors[0])
+        return None
 
-    def ancestors(self, filter_fn=None):
+    def ancestors(self: T, filter_fn: Callable[[T], bool]=None) \
```

Comment:
(Moving from a comment on a changeset)
Should this be `Optional[Callable[['DocCursor'], bool]]`?

Reply:
mypy actually infers the `Optional` from the fact that we're defaulting it to `None`, hooray for less verbosity!